indigo/sandbox/src/main/scala/com/example/sandbox/SandboxView.scala

```diff
@@ -88,21 +93,20 @@ object SandboxView {
         .withRef(Point(160, 120)),
       Graphic(0, 0, 320, 240, 1, SandboxAssets.lightMaterial.withTint(RGBA(1, 1, 0.0, 1)).withAlpha(1))
         .withRef(Point(160, 120))
-        .moveTo(inputState.mouse.position.x, inputState.mouse.position.y)
+        .moveTo(mouse.position.x, mouse.position.y)
     )
 
-  def uiLayer(inputState: InputState): List[SceneNode] =
+  def uiLayer(bl: BoundaryLocator): List[SceneNode] =
     List(
       Text("AB!\n!C", 2, 2, 5, Fonts.fontKey, SandboxAssets.fontMaterial.withAlpha(0.5)).alignLeft,
       Text("AB!\n!C", 100, 2, 5, Fonts.fontKey, SandboxAssets.fontMaterial.withAlpha(0.5)).alignCenter,
-      Text("AB!\n!C", 200, 2, 5, Fonts.fontKey, SandboxAssets.fontMaterial.withAlpha(0.5)).alignRight.onEvent {
-        case (bounds, _: MouseEvent.Click) =>
-          if (inputState.mouse.wasMouseClickedWithin(bounds))
-            println("Hit me!")
-          Nil
+      Text("AB!\n!C", 200, 2, 5, Fonts.fontKey, SandboxAssets.fontMaterial.withAlpha(0.5)).alignRight
+        .withEventHandler {
+          case (txt, MouseEvent.Click(pt)) if bl.bounds(txt).contains(pt) =>
+            println("Clicked me!")
+            None
```

Comment:
Nice 😄

Reply:
There is another form using `onEvent` which is a partial function from `(node, GlobalEvent) => GlobalEvent` that I hope will be used more as it's even cleaner. But if you want to result in no events and just `println` (which is only a debug use case really) then you do it this way.